flink/src/main/java/org/apache/iceberg/flink/source/DataIterator.java

```diff
@@ -41,16 +42,45 @@
   private final FileScanTaskReader<T> fileScanTaskReader;
 
   private final InputFilesDecryptor inputFilesDecryptor;
+  private final CombinedScanTask combinedTask;
+
   private Iterator<FileScanTask> tasks;
   private CloseableIterator<T> currentIterator;
+  private Position position;
 
   public DataIterator(FileScanTaskReader<T> fileScanTaskReader, CombinedScanTask task,
                       FileIO io, EncryptionManager encryption) {
     this.fileScanTaskReader = fileScanTaskReader;
 
     this.inputFilesDecryptor = new InputFilesDecryptor(task, io, encryption);
+    this.combinedTask = task;
+
     this.tasks = task.files().iterator();
     this.currentIterator = CloseableIterator.empty();
+    // fileOffset starts at -1 because we started
+    // from an empty iterator that is not from the split files.
+    this.position = new Position(-1L, 0L);
+  }
+
+  public void seek(Position startingPosition) {
+    // skip files
+    Preconditions.checkArgument(startingPosition.fileOffset() < combinedTask.files().size(),
+        "Checkpointed file offset is %d, while CombinedScanTask has %d files",
+        startingPosition.fileOffset(), combinedTask.files().size());
+    for (long i = 0L; i < startingPosition.fileOffset(); ++i) {
```

Comment:
Is `fileOffset()` a `long`? That seems odd to me. When would you need to address more than 2 billion files in a single combined scan task?

Reply:
integer would certainly be sufficient. I was using `long` to match the type in `RecordAndPosition` from flink-connector-files module. I have already tried to move away from depending on code from flink-connector-files. This looks like a miss due to the flink-table-planner-blink_2.12 pulling in flink-connector-files transitively.

The `long offset` from Flink's `RecordAndPosition` actually meant byte offset within a file. I will define our own `RecordAndPosition` and change `fileOffset` to `int` type.

Reply:
same as another comment. will update